File: Sorting/BubbleSort.cpp
#include<bits/stdc++.h>
#define MAX 100 
using namespace std;

void swap(int arr[], int x, int y){
	int temp = arr[x];
	arr[x] = arr[y];
	arr[y] = temp;
}
void bubbleSort(int arr[], int n){
		for(int i=0; i<n; i++){
		for(int j=0; j<n-i-1; j++){
			if(arr[j] > arr[j+1])
				swap(arr,j,j+1);
		}
	}
}
void displayArray(int arr[], int n){
	for(int i=0; i<n; i++){
		cout << arr[i] << " ";
	}

}
int main(){
	int arr[MAX],n=10;
	
	for(int i=0; i<n; i++){
		arr[i] = rand();
	}	
	cout << "Random Array" << endl;
	displayArray(arr, n);
	cout << endl;
	bubbleSort(arr, n);
	cout << "Sorted Array" << endl;
	displayArray(arr, n);
	
	return 0;
}

File: ALGORITHM in JavaScript/Sorting/BubbleSort.js
// 2 1 3 4 5

function swap(arr,x,y) {
    let temp = arr[x];
    arr[x] = arr[y];
    arr[y] = temp;
}

function bubbleSort(arr,n) {
    for(let i=0; i<n; i++){
        for(let j=0; j<n-i-1; j++){
            if(arr[j] > arr[j+1])
                swap(arr,j,j+1)
        }
    }
}


let arr = [64, 34, 25, 12, 22, 11, 90];
let n = arr.length;
bubbleSort(arr, n);
for(let i=0; i<n; i++){
    console.log(arr[i]);
}
